index.js

```diff
@@ -95,14 +96,18 @@ module.exports = (async (file, {readStart, writeStart} = {}) => {
         },
         flush(callback) {
             _writePromise
-                .then(() => { 
-					fs.truncate(fd, writeIndex);
-					close(fd);
-				})
+                .then(() => close(fd))
                 .then(callback);
         }
     });
 
+	
+	//Truncate the file up to where the write ends, otherwise
+	//some of the original content will remain in place
+	writeStream.on('finish', () => {
```

Comment:
Hi,

I'm not sure if it makes sense to move the truncate here. Writing may not yet be fully done. The prior option (truncate in flush) would be better, but it should be something like:

```javascript
_writePromise
.then(() => fs.ftruncate(fd, writeIndex))
.then(() => close(fd))
.then(() => callback())
.catch(e => callback(e))
```

This way we make sure the truncate runs AFTER the file is written and we ommit synchronous FS.

Reply:
Hey, I'll fix this during the weekend.

I'll need to add some test scenarios also.

Reply:
Hi, I agree. However, I had  noticed that **flush** is never called. I checked [ _stream_writeable.js ](https://github.com/nodejs/node/blob/64219741218aa87e259cf8257596073b8e747f0a/lib/_stream_writable.js) and I don't see any reference to **flush**.  I am working with node@v10.15.3.

Reply:
The only reference to **flush** in the docs is in the transform stream which is called when there is no more data to consume. I could be wrong but it looks like we should not be implementing flush in  the writeable as doing so has no effect!